lua/telescope/builtin/files.lua

```diff
@@ -186,30 +187,43 @@ files.current_buffer_fuzzy_find = function(opts)
 end
 
 files.tags = function(opts)
-  local ctags_file = opts.ctags_file or 'tags'
+  local tagfiles = opts.ctags_file and { opts.ctags_file } or
+                   vim.fn.map(vim.fn.tagfiles(), 'fnamemodify(v:val, ":p")')
 
-  if not vim.loop.fs_open(vim.fn.expand(ctags_file), "r", 438) then
-    print('Tags file does not exists. Create one with ctags -R')
+  if vim.tbl_isempty(tagfiles) then
+    print('No tags file found. Create one with ctags -R or verify your &tags option is correctly set.')
     return
   end
 
-  local fd = assert(vim.loop.fs_open(vim.fn.expand(ctags_file), "r", 438))
-  local stat = assert(vim.loop.fs_fstat(fd))
-  local data = assert(vim.loop.fs_read(fd, stat.size, 0))
-  assert(vim.loop.fs_close(fd))
+  finder = finders.new_coroutine()
+    -- opts.entry_maker or make_entry.gen_from_ctags(opts),
+    -- opts.max_results),
 
-  local results = vim.split(data, '\n')
+  local results = {}
+  for _, each in ipairs(tagfiles) do
+    local tags_directory = vim.fn.fnamemodify(each, ":h")
+    local resolve_filename = function(filename)
+        return path.exists(filename) and filename or
+               path.join(tags_directory, filename)
+    end
+    path.read(each, function(err, data)
```

Comment:
Hmmm... is path.read sync or async? Didn't read the rest.

If async, probably need to wait til it's complete before calling picker, or implement the streaming to add values.

Reply:
Async.